src/ipld/cid_hashmap.rs

```diff
@@ -22,6 +22,53 @@ pub struct CidHashMap<V> {
     fallback_hash_map: HashMap<Cid, V>,
 }
 
+impl<V> Extend<(Cid, V)> for CidHashMap<V> {
+    fn extend<T: IntoIterator<Item = (Cid, V)>>(&mut self, iter: T) {
+        for (k, v) in iter {
+            self.insert(k, v);
+        }
+    }
+}
+
+impl<V> FromIterator<(Cid, V)> for CidHashMap<V> {
+    fn from_iter<T: IntoIterator<Item = (Cid, V)>>(iter: T) -> Self {
+        let mut map = Self::new();
+        map.extend(iter);
+        map
+    }
+}
+
+pub struct IntoIter<V> {
+    small: std::collections::hash_map::IntoIter<[u8; BLAKE2B256_SIZE], V>,
+    fallback: std::collections::hash_map::IntoIter<Cid, V>,
+}
+
+impl<V> IntoIterator for CidHashMap<V> {
+    type Item = (Cid, V);
+    type IntoIter = IntoIter<V>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let CidHashMap {
+            v1_dagcbor_blake2b_hash_map,
+            fallback_hash_map,
+        } = self;
+        Self::IntoIter {
+            small: v1_dagcbor_blake2b_hash_map.into_iter(),
+            fallback: fallback_hash_map.into_iter(),
+        }
+    }
+}
+
+impl<V> Iterator for IntoIter<V> {
+    type Item = (Cid, V);
+    fn next(&mut self) -> Option<Self::Item> {
+        self.small
+            .next()
+            .map(|(bytes, v)| (Cid::from(CidVariant::V1DagCborBlake2b(bytes)), v))
+            .or_else(|| self.fallback.next().map(|(cid, v)| (cid, v)))
```

Comment:
This is the same, right?

```suggestion
            .or_else(|| self.fallback.next())
```